ports/geckolib/values.rs

```diff
@@ -159,14 +159,16 @@ impl GeckoStyleCoordConvertible for LengthOrPercentage {
     fn to_gecko_style_coord(&self, unit: &mut nsStyleUnit, union: &mut nsStyleUnion) {
         match *self {
             LengthOrPercentage::Length(au) => {
+                unsafe { union.reset(unit) };
```

Comment:
Perhaps I should move these reset calls out and remove the reset from set_calc_value. I designed it this way initially because set_calc_value was calling `nsStyleCoord::SetCalcValue()` which resets, but now it's a manual impl so this restriction doesn't apply anymore.

(I switched to a manual impl because in stylo we deal with `union,unit` pairs more than directly dealing with nsStyleCoord)